coalib/settings/Setting.py

```diff
@@ -163,7 +163,11 @@ def __path__(self, origin=None, glob_escape_origin=False):
         :raises ValueError:        If no origin is specified in the setting
                                    nor the given origin parameter.
         """
-        strrep = str(self).strip()
+        if hasattr(self, 'value'):
+            strrep = self.value.strip()
```

Comment:
> I think self will always have value and the else path will never be taken.

I thought the same too, but ... we have `str` objects being passed in as self, as crazy as that sounds.

If the `else` path wasnt being taken, the 100% coverage would fail :P 

I've tested adding a `Setting.__str__`, and that fails oh so many tests in `coala`, and PyLintBear fails also.  Nice idea though; worth a shot.

I can replace `if hasattr(self, 'value'):` with `if isinstance(self, Setting):` , which makes the bizarre situation a bit more obvious.  I've got no idea what madness has created this situation; I am sure when I figure it out, I will be hell bent on eradicating it.